.circleci/config.yml

```diff
@@ -0,0 +1,145 @@
+version: 2.1
+references:
+  install_prerequisites: &install_prerequisites
+    run:
+        name: Install Prerequisites
+        command: |
+          mkdir -p $ROS_WS/src
+          vcs import $ROS_WS/src < $NAV2_WS/src/navigation2/tools/ros2_dependencies.repos
+          find $ROS_WS/src -not -iwholename '*.git/*' -type f -exec sha256sum {} \; | \
+            sort -k 2 | \
+            cut -d" " -f1 | \
+            sha256sum | \
+            awk '{print "source "$1}' >> $ROS_WS/checksum.txt
+          apt-get update
+          dependencies=$(
+            rosdep install -q -y \
+              --from-paths \
+                $ROS_WS/src \
+              --ignore-src \
+              --verbose | \
+            awk '$1 ~ /^resolution\:/' | \
+            awk -F'[][]' '{print $2}' | \
+            tr -d \, | xargs -n1 | sort -u | xargs)
+          dpkg -s $dependencies | \
+            sha256sum | \
+            awk '{print "dependencies "$1}' >> $ROS_WS/checksum.txt
```

Comment:
Looking at this again, I'm not sure this alone would be a sufficient cache break. It may likely be that core ros packages installed by the nightly flat archives from the base image could change; and as they aren't debs, listing from dpkg would not generate a diff. Perhaps we could append a digest of `/opt/ros/$ROS_DISTRO/` to `$ROS_WS/checksum.txt`?

This would also essentially break the cache after every new CI docker image update, but I don't spot an immediate workaround. How often are the prerequisites in the `ros2_dependencies.repos` changed in PRs? We could comment out the prerequisites steps in the build job, and let the CI docker image keep them uptodate; and if someone does need to test a change there, they only need to uncomment lines L119 and L121.

Reply:
Resolved with https://github.com/ros-planning/navigation2/pull/534#issuecomment-459842619